lib/logstash/outputs/elasticsearch.rb

```diff
@@ -557,9 +554,30 @@ def common_event_params(event)
     #      }
     params[:pipeline] = target_pipeline unless (target_pipeline.nil? || target_pipeline.empty?)
 
+    params[:version] = resolve_version(event, event_version)
+    params[:version_type] = resolve_version_type(event, event_version_type)
+
     params
```

Comment:
The issues raised elsewhere could likely be resolved by leaning on [`Hash#compact`](https://ruby-doc.org/core-2.7.1/Hash.html#method-i-compact), which is available [since Ruby 2.4](https://apidock.com/ruby/v2_4_6/Hash/compact) (and therefore on all Logstash 7+):

```suggestion
    params.compact
```